daemon/algod/api/server/v2/handlers.go

```diff
@@ -443,6 +501,40 @@ func (v2 *Handlers) getPendingTransactions(ctx echo.Context, max *uint64, format
 	return ctx.Blob(http.StatusOK, contentType, data)
 }
 
+// startCatchup Given a catchpoint, it starts catching up to this catchpoint
+func (v2 *Handlers) startCatchup(ctx echo.Context, catchpoint string) error {
+	_, _, err := ledger.ParseCatchpointLabel(catchpoint)
+	if err != nil {
+		return badRequest(ctx, err, errFailedToParseCatchpoint, v2.Log)
+	}
+
+	err = v2.Node.StartCatchup(catchpoint)
+	if err != nil {
+		return internalError(ctx, err, fmt.Sprintf(errFailedToStartCatchup, err), v2.Log)
+	}
+
+	return ctx.JSON(http.StatusOK, private.CatchpointStartResponse{
+		CatchupMessage: catchpoint,
+	})
+}
+
+// abortCatchup Given a catchpoint, it aborts catching up to this catchpoint
+func (v2 *Handlers) abortCatchup(ctx echo.Context, catchpoint string) error {
+	_, _, err := ledger.ParseCatchpointLabel(catchpoint)
+	if err != nil {
+		return badRequest(ctx, err, errFailedToParseCatchpoint, v2.Log)
+	}
+
+	err = v2.Node.AbortCatchup(catchpoint)
+	if err != nil {
+		return internalError(ctx, err, fmt.Sprintf(errFailedToAbortCatchup, err), v2.Log)
+	}
+
+	return ctx.JSON(http.StatusOK, private.CatchpointAbortResponse{
+		CatchupMessage: catchpoint,
+	})
+}
```

Comment:
These two, including the Shutdown call are not RESTful endpoints. Given these private endpoints grow more than anticipated, we should consider switching to a separate thing Maybe gRPC.

Reply:
I would be glad to do so, but not in the context of this PR. Preferably, we would have a PR going in ahead of this one, which would introduce the gRPC framework and having this PR extends that functionality.